### PR TITLE
User/ endpoint work & Stub remaining endpoints

### PR DESCRIPTION
**Pull request by tsa96**
Completes most stuff in user/.

Then adds controller and DTO layer for all the remaining endpoints I could find besides maps/ (Alex) and some remaining on auth/ (me, next).

This should make life a lot easier for our new devs having an initial layer in place, requires referring back to old API much less.

### CODE REVIEW

File: server/src/@common/dto/map/map-notify.dto.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿import { ApiProperty, PickType } from '@nestjs/swagger';
+import { MapNotify } from '@prisma/client';
+import { Exclude } from 'class-transformer';
+
+export class MapNotifyDto implements MapNotify {
```
Comment (TheNerdMan) [inline server/src/@common/dto/map/map-notify.dto.ts:5]: Missing properties that are on the Prisma model

File: server/src/@common/dto/map/createMap.dto.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { ApiProperty } from '@nestjs/swagger';
 import { EMapTriggerType, EMapType } from '../../enums/map.enum';
-import { MapCreditDto } from './mapCredit.dto';
+import { MapCreditDto } from './map-credit.dto';
 
+// TODO: Alex, these are really messy, can they be moved to their respective DTO files?
```
Comment (TheNerdMan) [inline server/src/@common/dto/map/createMap.dto.ts:5]: Yep yep, this was just temp while I got the tests working

File: server/src/@common/filters/all-exceptions.filter.ts [hunk 3]
```diff
@@ -30,7 +31,7 @@ export class AllExceptionsFilter implements ExceptionFilter {
             e
         );
 
-        if (!(e instanceof UnauthorizedException) && !(e instanceof NotFoundException)) {
+        if (e instanceof InternalServerErrorException) {
```
Comment (TheNerdMan) [inline server/src/@common/filters/all-exceptions.filter.ts:34]: Has this been tested that it catches all errors appart from 401 and 404?

File: server/src/@common/dto/map/map-notify.dto.ts (hunk 1 above)
Comment (tsa96) [reply server/src/@common/dto/map/map-notify.dto.ts:5]: I think your Prisma client is probably out of date, the current type is
```ts
export type MapNotify = {
  notifyOn: number
  mapID: number
  userID: number
  createdAt: Date
  updatedAt: Date
}
```

File: server/src/@common/filters/all-exceptions.filter.ts (hunk 3 above)
Comment (tsa96) [reply server/src/@common/filters/all-exceptions.filter.ts:34]: We don't want it catching any of the 400s. Some other 500s we should probably get, I'll come back to this

**Review by Gocnak — APPROVED** [review]

**Pull request merged**